### PR TITLE
feat: analyze resume

### PR DESCRIPTION
**Pull request by Linchen-Xu**
## Context

> Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Changes

> Please list your changes here. For example:
>- Add analyze button

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: fe_repo/src/functions/api.ts [hunk 1]
```diff
@@ -25,6 +25,18 @@ export function sendMessage(message: string) {
   }
 }
 
+export function analyze(job_description: string) {
```
Comment (adamma1024) [inline fe_repo/src/functions/api.ts:28]: One suggestion, could you also add test cases for any functions in this file?

File: fe_repo/src/pages/ChatBox.tsx [hunk 2]
```diff
@@ -20,53 +23,136 @@ export const ChatBox = ({onSendMessage}: IChatBoxProps) => {
...
+    backgroundColor: 'white',
+    padding: '20px',
+    borderRadius: '8px',
+    boxShadow: '0px 4px 8px rgba(0, 0, 0, 0.1)',
+    zIndex: 1000,
+    width: '400px',
+    display: showPrompt ? 'block' : 'none', // Ensure the value of 'display' is a valid CSS value
+  };
+
+  const overlayStyle: CSSProperties = {
```
Comment (adamma1024) [inline fe_repo/src/pages/ChatBox.tsx]: Awesome, but there is a suggestion for styles, I imported the tailwindcss as our CSS lib. It's better to transform these styles to the tailwind defined class name. For example, `top: "50%"`, `left: "50%"` equals to `top-1/2`, `left-1/2`
You can find all supported styles here: https://tailwindcss.com/docs/top-right-bottom-left
In addition, you can ask GPT for tailwind css directly.

image

File: fe_repo/src/pages/ChatBox.tsx [hunk 3]
```diff
@@ -20,53 +23,116 @@ export const ChatBox = ({onSendMessage}: IChatBoxProps) => {
...
+  const handleAnalyze = () => {
+    if (jobDescription.trim() !== "") {
+      // Call the analyze function with job description
+      onAnalyze(jobDescription);
+    }
+    setShowPrompt(false);
+    setJobDescription(""); // Clear the input after submission
+  };
+
+  const iconRowStyle = {
```
Comment (adamma1024) [inline fe_repo/src/pages/ChatBox.tsx]: And these codes

File: fe_repo/src/pages/ChatBox.tsx [hunk 4]
```diff
@@ -20,53 +23,116 @@ export const ChatBox = ({onSendMessage}: IChatBoxProps) => {
...
+    border: 'none',
+    fontSize: '14px', // Ensure consistent font size
     display: 'flex',
-    alignItems: 'center' as const as 'center',
-    backgroundColor: '#ffffff', // White background for chat input
-    borderRadius: '8px',
-    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.1)',
+    alignItems: 'center',
+    gap: '5px', // Space between icon and text
   };
```
Comment (adamma1024) [inline fe_repo/src/pages/ChatBox.tsx]: These styles

**Comment by Linchen-Xu** [comment]
All resolved, including other declarations in all files. PTAL @adamma1024

**Review by adamma1024 — APPROVED** [review]

**Pull request merged**